### DISCUSSION TITLE
Get cert of remote host in tunnel?

### DISCUSSION THREAD

**Discussion by samwell-ho** [strongswan/strongswan#2390] [Q&A]
I am trying to set up an IPSec system with strongSwan, which includes many points connected to each other in a mesh tunnel.

The problem is: Each point needs its cert, key, CA's cert and the cert of the remote host it needs to create the tunnel. This is quite complicated in the key distribution problem.

I want to ask if there is a way to set up so that each point only needs its cert and key and CA's cert. The cert of the remote host will be received from the remote host that it creates the tunnel.

What does the keyword "left|rightsendcert = never | no | ifasked | always | yes" mean in this case?

**Comment by tobiasbrunner** [comment]
What makes you think you need to configure the remote peers' certificates explicitly?

**Comment by samwell-ho** [comment]
The reason is that in an existing system (including hundreds of hosts), if I add a new host, I have to update the cert for the remaining hosts. This is quite inconvenient.

**Comment by samwell-ho** [comment]
I tried configuring a tunnel, if "rightcert=..." is not clearly declared, it cannot be up.

Maybe there is something missing in the tunnel configuration, please help me.

**Reply by tobiasbrunner** [reply]
Since that's not generally necessary, yes your config or PKI is probably missing something. But since you haven't posted any configs, logs or information about your certificates (e.g. `ipsec listcerts`) there is really nothing we can do here (see #196).
